lib/firestore/firestore.dart

```diff
@@ -380,3 +379,93 @@ Future<Map<dynamic, dynamic>>? retrieveAppointment(id) async {
   });
   return appointmentDetails;
 }
+
+Future<List<dynamic>> retrieveMessagesPhysicians(uid) async {  
+  Map<dynamic, dynamic> physician = await db
+      .collection("Practitioner")
+      .doc(uid)
+      .get()
+      .then((DocumentSnapshot doc) {
+    return doc.data() as Map<String, dynamic>;
+  }, onError: (e) => debugPrint("Error getting document: $e"));
+
+  QuerySnapshot chats = await db
+      .collection("Chats")
+      .where('user2', isEqualTo: physician["id"])
+      .get();
+
+  final allData = [];
+  for (var chatDoc in chats.docs) {
+    Map<String, dynamic> chat =
+        chatDoc.data() as Map<String, dynamic>;
+    allData.add(chat);
+  }
+  return allData;
+}
+Future<List<dynamic>> retrieveMessagesPatients(uid) async {  
+  Map<dynamic, dynamic> patient = await db
+      .collection("Patient")
+      .doc(uid)
+      .get()
+      .then((DocumentSnapshot doc) {
+    return doc.data() as Map<String, dynamic>;
+  }, onError: (e) => debugPrint("Error getting document: $e"));
+
+  QuerySnapshot chats = await db
+      .collection("Chats")
+      .where('user1', isEqualTo: patient["id"])
+      .get();
+
+  final allData = [];
+  for (var chatDoc in chats.docs) {
+    Map<String, dynamic> chat =
+        chatDoc.data() as Map<String, dynamic>;
+    allData.add(chat);
+  }
+  return allData;
+}
+
+Future<Map<String, dynamic>>? retrieveChatCreationInfo(uid) async {
+  //retrieving patients
+  final pracRef = db.collection("Practitioner").doc(uid);
+
+  Map<dynamic, dynamic> list = await pracRef.get().then((DocumentSnapshot doc) {
+    return doc.data() as Map<String, dynamic>;
+  }, onError: (e) => debugPrint("Error getting document: $e"));
+
+  String physID = list["id"];
+
+  // parsing patient list
+  List<dynamic> data = list["patients"];
+  List<PatientID> patients = data.map((e) {
+    Gender gender = e["gender"] == "M" ? Gender.male : Gender.female;
+    Timestamp t = e["birthdate"] as Timestamp;
+    DateTime birthdate = t.toDate();
+
+    return PatientID(e["name"], gender, e["id"], birthdate);
+  }).toList();
+
+  Map<String, dynamic> map = {};
+  map["patients"] = patients;
+  return map;
+}
+
+void createConversation(Map<String, dynamic> payload, String? uid) async {
+
+  final docRef = db.collection("Chats").doc();
+  final docRefMessage = db.collection("Chats").doc(docRef.id).collection("Messages");
+  Physician phys = await retrievePhysicianProfile(uid);
+  DateTime date = DateTime.now();
+  Timestamp lastSent = Timestamp.fromDate(date);
+  await docRef.set({
+    "appointmentID": docRef.id,
+    "patient": payload["patient"],
+    "practitioner": phys.id,
+    "appointmentType": payload["type"],
+    "description": payload["description"],
+    "created": DateTime.now(),
+    "serviceCategory": "appointment",
+    "scheduledTimeStart": payload["scheduledTimeStart"],
+    "scheduledTimeEnd": payload["scheduledTimeEnd"],
+  });
```

Comment:
Is this still to be changed? It doesn't match what you have setup in the database for the actual messages.

Reply:
ah my thing did not correctly transfer over gimme one sec im gonna make  another pull request then